plugins/modules/pfsense_virtual_ip.py

```diff
@@ -0,0 +1,316 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2018, Frederic Bor <frederic.bor@wanadoo.fr>
+# Copyright: (c) 2021, Jan Wenzel <jan.wenzel@gonicus.de>
+# Copyright: (c) 2023, Martin Müller <martin.mueller@dataport.de>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = """
+---
+module: pfsense_virtual_ip
+version_added: "0.6.2"
+author: Jan Wenzel (@coffeelover)
+short_description: Manage pfSense virtual ip settings
+description:
+  - Manage pfSense virtual ip settings
+notes:
+options:
+  mode:
+    description: Type
+    required: True
+    type: str
+    choices: [ "proxyarp", "carp", "ipalias", "other" ]
+  descr:
+    description: Description
+    required: False
+    type: str
+  interface:
+    description: Interface
+    required: True
+    type: str
+  vhid:
+    description: VHID Group
+    required: False
+    type: int
+  advbase:
+    description: Advertising Frequency Base
+    required: False
+    type: int
+  advskew:
+    description: Advertising Frequency Skew
+    required: False
+    type: int
+  password:
+    description: Virtual IP Password
+    required: False
+    type: str
+  uniqid:
+    description: Unique ID of Virtual IP in configuration
+    required: False
+    type: str
+  type:
+    description: Address Type
+    required: False
+    type: str
+    choices: [ "single" ]
+    default: single
+  subnet_bits:
+    description: Network's subnet mask
+    required: False
+    type: int
+    default: 32
+  subnet:
+    description: Network subnet
+    required: False
+    type: str
+  state:
+    description: State in which to leave the Virtual IP
+    choices: [ "present", "absent" ]
+    default: present
+    type: str
+"""
+
+EXAMPLES = """
+- name: Setup Home vip
+  pfsense_virtual_ip:
+    mode: "carp"
+    descr: "HOME VIP"
+    interface: "opt2"
+    vhid: 24
+    advbase: 1,
+    advskew: 0,
+    password": "xaequae0sheiB7sh"
+    uniqid": "vip_home"
+    subnet_bits": 24
+    subnet": "10.1.1.1"
+    state": "present"
+"""
+
+RETURN = """
+"""
+
+import re
+from copy import deepcopy
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.pfsensible.core.plugins.module_utils.module_base import PFSenseModuleBase
+
+
+VIRTUALIP_ARGUMENT_SPEC = dict(
+    mode=dict(required=True, choices=['proxyarp', 'carp', 'ipalias', 'other'], type='str'),
```

Comment:
I'm not sure this should be required - is it needed for the state = absent case?

Reply:
Me too. Some code is not from me - ping @coffeelover 

(And I do not touch that firewall part very often)

Reply:
IIRC a mode is required or at least one mode should be default (i would prefer ipalias).
Also for state=absent, mode is required, because only for carp and ipalias the virtual interface has to be manually removed.